python/lsst/ctrl/bps/bps_utils.py

```diff
@@ -145,10 +145,16 @@ def _create_execution_butler(config, qgraph_filename, execution_butler_dir, out_
     out_filename = "execution_butler_creation.out"
     if out_prefix is not None:
         out_filename = os.path.join(out_prefix, out_filename)
-    with open(out_filename, "w") as fh:
-        print(command, file=fh)
-        print("\n", file=fh)  # Note: want a blank line
-        subprocess.run(shlex.split(command), shell=False, check=True, stdout=fh, stderr=subprocess.STDOUT)
+    opening = "cannot create the execution Butler"
+    try:
+        with open(out_filename, "w") as fh:
+            print(command, file=fh)
+            print("\n", file=fh)  # Note: want a blank line
+            subprocess.run(shlex.split(command), shell=False, check=True, stdout=fh, stderr=subprocess.STDOUT)
+    except OSError as exc:
+        raise type(exc)(f"{opening}: {exc.strerror}") from None
```

Comment:
Could be helpful to have a brief comment here.  It took me a minute to figure out why doing separate catching of the the exceptions.